prov/efa/src/rxr/rxr_init.c

```diff
@@ -591,9 +595,14 @@ static int rxr_getinfo(uint32_t version, const char *node,
 			assert(!strcmp(shm_info->fabric_attr->name, "shm"));
 		}
 	}
-out:
+
 	fi_freeinfo(core_info);
 	return ret;
+free_info:
+	fi_freeinfo(util_info);
+	fi_freeinfo(*info);
+	*info = NULL;
```

Comment:
do we need ```free(core_info)``` here?

Reply:
Yep, you're right, fixed. Looks like that was missing previously in the error paths.